src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1927,6 +1927,7 @@ internal ImmutableArray<Diagnostic> GetDiagnostics(CompilationStage stage, bool
                             var syntaxTree = syntaxTrees[i];
                             AppendLoadDirectiveDiagnostics(builder, _syntaxAndDeclarations, syntaxTree);
                             builder.AddRange(syntaxTree.GetDiagnostics(cancellationToken));
+                            builder.AddRange(syntaxTree.Options.Errors);
```

Comment:
I should add that it makes sense for the syntax option diagnostics to be reported here for the same reason that it makes sense for the syntax diagnostics to be reported here. However, the syntax option diagnostics should not be reported repeatedly for the same syntax options. When the same options are used for multiple trees (which is the normal case) they should be reported once per options object.